### PR TITLE
Homework9: Ekaterina Tsiunel

### PR DESCRIPTION
**Pull request by ETsiunel**

### CODE REVIEW

**Comment by ETsiunel** [comment]
@Stml89 PR is ready for review

File: hw_9.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+# # # Строки с заданным символом # # #
+# Напишите программу, которая бы работала следующим образом -
+# находила символ "#" и если этот символ найден -
+# удаляла предыдущий символ из строки.
+# Ваша задача обработать строки с "#" символом.
+
+
+def string_modifier(string: str) -> str:
+    """Функция замены символа перед #"""
+    if "".join(string.split("#")) == "":
```
Comment (Stml89) [inline hw_9.py]:
```suggestion
    if not "#" in string:
```

File: hw_9.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+# сожгите 5 свечей, получите 5 остатков;
+# создайте еще 2 свечи, используя 4 остатка (остался 1);
+# сожгите 2 свечи, в итоге останется 3 остатка;
+# создайте еще одну свечу, используя 2 остатка (остался 1);
+# сожгите созданную свечу, что даст еще один остаток (всего 2 остатка);
+# создать свечу из оставшихся остатков;
+# зажгите последнюю свечу.
+# Таким образом, можно сжечь 5+2+1+1=9 свечей, что и является ответом.
+new_candles_init = 0
+make_new = 0
```
Comment (Stml89) [inline hw_9.py]: where are you using these 2 variables?

File: hw_9.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+
+def burned_candles_counter(new_candles, make_new_candle):
```
Comment (Stml89) [inline hw_9.py:56]: if you decided yo use typehints, pls use it everywhere

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw_9.py (hunk 1 above)
Comment (ETsiunel) [reply hw_9.py]: `    if not "#" in string:`
It does not work in case of string "abcde"

File: hw_9.py (hunk 2 above)
Comment (ETsiunel) [reply hw_9.py]: nowhere, fixed

File: hw_9.py (hunk 3 above)
Comment (ETsiunel) [reply hw_9.py:56]: Sorry, it's not clear what you mean: are you about variable names?

File: hw_9.py (hunk 1 above)
Comment (Stml89) [reply hw_9.py]: Just return the string

File: hw_9.py (hunk 3 above)
Comment (Stml89) [reply hw_9.py:56]: `def burned_candles_counter(new_candles: <type>, make_new_candle: <type>) -> <type>:`

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: hw_9.py (hunk 3 above)
Comment (ETsiunel) [reply hw_9.py:56]: fixed

File: hw_9.py (hunk 1 above)
Comment (ETsiunel) [reply hw_9.py]: fixed

**Comment by ETsiunel** [comment]
All comments fixed @Stml89

**Review by Stml89 — APPROVED** [review]

**Pull request merged**